modules/top-level/plugins/utils.nix

```diff
@@ -0,0 +1,38 @@
+lib:
+lib.fix (self: {
+  normalizedPluginType = lib.types.submodule {
+    options = {
+      plugin = lib.mkOption {
+        type = lib.types.package;
+      };
+
+      config = lib.mkOption {
+        type = with lib.types; nullOr str;
+      };
+
+      optional = lib.mkOption {
+        type = lib.types.bool;
+      };
+    };
+  };
+
+  # Normalize a plugin in a standard { plugin, config, optional } attrs
+  normalizePlugin =
```

Comment:
If we're going to use a submodule-type for normalizedPluginType, could we consider not having a "normalize plugins" function and instead rely on the module system (defaults, etc) to do the nomalisation for us?

Reply:
Let's hold off and consider it in another PR? As I think we use `normalizePlugin` and `normalizePlugins` in a handful of unrelated places.